src/pages/workspace/AccessOrNotFoundWrapper.tsx

```diff
@@ -0,0 +1,109 @@
+/* eslint-disable rulesdir/no-negated-variables */
+import React, {useEffect} from 'react';
+import type {OnyxEntry} from 'react-native-onyx';
+import {withOnyx} from 'react-native-onyx';
+import FullPageNotFoundView from '@components/BlockingViews/FullPageNotFoundView';
+import FullscreenLoadingIndicator from '@components/FullscreenLoadingIndicator';
+import Navigation from '@libs/Navigation/Navigation';
+import * as PolicyUtils from '@libs/PolicyUtils';
+import NotFoundPage from '@pages/ErrorPage/NotFoundPage';
+import * as Policy from '@userActions/Policy';
+import ONYXKEYS from '@src/ONYXKEYS';
+import ROUTES from '@src/ROUTES';
+import type * as OnyxTypes from '@src/types/onyx';
+import type {PolicyFeatureName} from '@src/types/onyx/Policy';
+import callOrReturn from '@src/types/utils/callOrReturn';
+import {isEmptyObject} from '@src/types/utils/EmptyObject';
+
+const POLICY_ACCESS_VARIANTS = {
+    PAID: (policy: OnyxEntry<OnyxTypes.Policy>) => !PolicyUtils.isPaidGroupPolicy(policy) || !policy?.isPolicyExpenseChatEnabled,
+    ADMIN: (policy: OnyxEntry<OnyxTypes.Policy>) => !PolicyUtils.isPolicyAdmin(policy),
+} as const satisfies Record<string, (policy: OnyxTypes.Policy) => boolean>;
```

Comment:
Also, I think using accessVariant name as a const (`CONST.POLICY.ACCESS_VARIANTS.PAID`) will be more safe than string (`'PAID'`)
```javascript
const POLICY_ACCESS_VARIANTS = {
    [CONST.POLICY.ACCESS_VARIANTS.PAID]: ....,
    [CONST.POLICY.ACCESS_VARIANTS.ADMIN]: ....
}
<AccessOrNotFoundWrapper accessVariants={[CONST.POLICY.ACCESS_VARIANTS.PAID]}

```

Reply:
1. Sure, I'll apply this, it seems more readable
2. the idea is that this const is being used only in the context of AccessOrNotFoundWrapper, so splitting its keys to a separate file might be unintuitive, especially since `POLICY_ACCESS_VARIANTS` is being used as the type source which would be exported to the other places. I'd suggest keeping it this way until we find a case for using `POLICY_ACCESS_VARIANTS` outside of this component and then move it as a whole

Reply:
> the idea is that this const is being used only in the context of AccessOrNotFoundWrapper.

@BrtqKr But it used in many places where `AccessOrNotFoundWrapper` is used as a prop of it, so I think we need to update it.

Also, this point is a Reviewer Checklist item
> [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such